docs/proposals/thvignore.md

```diff
@@ -0,0 +1,346 @@
+
+     * Determine full container path (e.g. `/app/.ssh`)
+
+     * Add a `tmpfs` mount over it to the runtime configuration
+
+---
+
+## **🧱 Component Design**
+
+### **🔹 `IgnoreProcessor` (new module in `pkg/ignore/`)**
+
+```go
+package ignore
+
+type IgnoreProcessor struct {
+    GlobalPatterns []string
+    LocalPatterns  []string
+}
+
+func NewIgnoreProcessor() *IgnoreProcessor
+func (ip *IgnoreProcessor) LoadGlobal() error
+func (ip *IgnoreProcessor) LoadLocal(sourceDir string) error
+func (ip *IgnoreProcessor) GetOverlayPaths(bindMount, containerPath string) []string
+func (ip *IgnoreProcessor) ShouldIgnore(path string) bool
+```
+
+*
+  Reads `.gitignore`\-style files using existing Go libraries
+
+* Integrates with ToolHive's existing mount processing pipeline
+
+* Converts ignore patterns into **container absolute paths** (e.g. `/app/.ssh`)
+
+---
+
+### **🔹 Enhanced `runtime.Mount` (in `pkg/container/runtime/types.go`)**
+
+Extend the existing Mount struct to support tmpfs:
+
+```go
+type Mount struct {
+    Source   string
+    Target   string
+    ReadOnly bool
+    Type     string // NEW: "bind" or "tmpfs"
+}
+```
+
+Integration with existing mount processing in `pkg/runner/config_builder.go`:
+
+```go
+func (b *RunConfigBuilder) processVolumeMounts() error {
+    // Existing mount processing...
+    
+    // NEW: Process ignore patterns
+    ignoreProcessor := ignore.NewIgnoreProcessor()
+    ignoreProcessor.LoadGlobal()
+    ignoreProcessor.LoadLocal(sourceDir)
+    
+    overlayPaths := ignoreProcessor.GetOverlayPaths(source, target)
+    for _, overlayPath := range overlayPaths {
+        b.addTmpfsOverlay(overlayPath)
+    }
+}
+```
+
+---
+
+## **🧪 Runtime Support**
+
+Enhanced `convertMounts` function in `pkg/container/docker/client.go`:
+
+```go
+func convertMounts(mounts []runtime.Mount) []mount.Mount {
+    result := make([]mount.Mount, 0, len(mounts))
+    for _, m := range mounts {
+        if m.Type == "tmpfs" {
+            result = append(result, mount.Mount{
+                Type:   mount.TypeTmpfs,
+                Target: m.Target,
+                TmpfsOptions: &mount.TmpfsOptions{
+                    SizeBytes: 1024 * 1024, // 1MB tmpfs for security overlays
+                },
+            })
+        } else {
+            result = append(result, mount.Mount{
+                Type:     mount.TypeBind,
+                Source:   m.Source,
+                Target:   m.Target,
+                ReadOnly: m.ReadOnly,
+            })
+        }
+    }
+    return result
+}
+```
+
+| Runtime | Bind Mount | Tmpfs Overlay |
+| ----- | ----- | ----- |
+| Docker | ✅ `mount.TypeBind` | ✅ `mount.TypeTmpfs` |
+| Podman | ✅ `--mount type=bind` | ✅ `--mount type=tmpfs` |
+
+---
+
+## **🧰 CLI Integration**
+
+Extend existing `thv run` command flags:
+
+```go
+// In cmd/thv/app/run.go
+var (
+    runIgnoreGlobally     bool
+    runPrintOverlays      bool
+    runIgnoreFile         string
+)
+
+func init() {
```

Comment:
maybe a useful command should be "ListIgnoredPatterns", that will show all patterns that are currently ignored in our envs (globally or per-file)